### PR TITLE
feat: home screen bottom sheet

### LINKED ISSUE #50: Create site from home screen sites list view
Depends on #33, #52

### LINKED ISSUE #52: Create home screen sites list view
If done before techmatters/terraso-client-shared#17, create an issue to wire it up to the API once techmatters/terraso-client-shared#17 is done.

### PR DESCRIPTION
**Pull request by shrouxm**
## Description
Adds a bottom sheet to the home screen.

### Checklist
- [x] Corresponding issue has been opened

### Related Issues
Closes #50 
Closes #52

### Verification steps
Sites should be visible in the bottom sheet. Clicking "show on map" should pan the map to the site. There is no location dashboard yet so clicking the site does nothing. Also updates the site map to not persist temporary sites, so there is at most one active at a time.

### CODE REVIEW

**Comment by shrouxm** [comment]
@paulschreiber can you run `pod install`? i think that is the only necessary iOS install step for the dependencies i added

File: dev-client/src/auth/index.ts [hunk 1]
```diff
@@ -27,6 +27,7 @@ export async function exchangeToken(
     headers: {'content-type': 'application/json'},
   });
 
+  console.log(payload);
```
Comment (paulschreiber) [inline dev-client/src/auth/index.ts:30]:
```suggestion
```

File: dev-client/src/components/home/BottomSheet.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+          <Heading size="xl">Sites</Heading>
+          <Button size="lg" onPress={addSiteCallback}>
+            CREATE SITE
+          </Button>
+        </Flex>
+        {siteList.length === 0 && (
+          <>
+            <Text fontSize="md">
+              You don’t currently have any saved sites. Create a site by using
+              the button above or by finding the site location on the map.
```
Comment (paulschreiber) [inline dev-client/src/components/home/BottomSheet.tsx:76]: issue: text should be localized.

File: dev-client/src/components/home/BottomSheet.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            </Text>
+            <Text />
+            <Text fontSize="md">
+              A site allows you to collect data, learn more about your land, and
+              share with others. Learn more about using LandPKS...
```
Comment (paulschreiber) [inline dev-client/src/components/home/BottomSheet.tsx:81]: issue: text should be localized.

**Review by paulschreiber — CHANGES_REQUESTED** [review]

File: dev-client/src/components/home/BottomSheet.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+          <Heading size="xl">{t('site.list_title')}</Heading>
+          <Button size="lg" onPress={addSiteCallback}>
+            {t('site.create')}
+          </Button>
+        </Flex>
+        {siteList.length === 0 && (
+          <>
+            <Text fontSize="md">{t('site.none_existing_p1')}</Text>
+            <Text />
+            <Text fontSize="md">{t('site.none_existing_p2')}</Text>
```
Comment (paulschreiber) [inline dev-client/src/components/home/BottomSheet.tsx:96]: question what do `none_existing_p1` and `none_existing_p2` mean? Can we give them more descriptive names? If these are just different paragraphs for the same message, can we use <Trans> to consolidate strings?

Comment (shrouxm) [reply dev-client/src/components/home/BottomSheet.tsx:96]: they are two paragraphs of the same message. i couldn't find a way to get it to render properly when it was all one string, but that was before i i18n'd it so maybe there is a way i will check

Comment (shrouxm) [reply dev-client/src/components/home/BottomSheet.tsx]: done

**Pull request merged**